### PR TITLE
Refactor category show

### PR DESCRIPTION
**Pull request by liambarstad**
LIST OF CHANGES:
- CSS styling on show page moved and renamed in categories.css, required in application.css
- Category index page shows categories by popularity
- Category index page renders images of most popular items in each category
- Item.by_popularity added

THINGS I'M UNSURE ON:
- Item popularity is not based on revenue because line_item_totals are stored as decimals in the database (ex. 0.16e2 = 16.00). From what I can deduce, this is preventing .order("sum(order_items.line_item_total) DESC") from working correctly
- e.g. the sum of 0.4e1 and 0.8e1 will be greater than 0.16e2

RSPEC GREEN?
YES

*MENTION GROUP MEMBER TO REVIEW PR*
@mimilettd @jimszalew @vaidashi

### CODE REVIEW

**Comment by jmejia** [comment]
@liambarstad this PR is MASSIVE. Please work in smaller chunks. Generally this is going to make it difficult to merge.

File: app/assets/javascripts/categories.coffee [hunk 1]
```diff
@@ -1,3 +1,15 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+
+# $(document).ready ->
+#   $('#cycle_forwards').on 'click', (event) ->
```
Comment (jmejia) [inline app/assets/javascripts/categories.coffee:6]: Why comment out and not delete?

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 class CategoriesController < ApplicationController
 
   def index
-    
+    @categories = Category.by_popularity
```
Comment (jmejia) [inline app/controllers/categories_controller.rb:4]: Nice abstraction

File: app/models/category.rb [hunk 3]
```diff
@@ -6,4 +6,20 @@ class Category < ApplicationRecord
   include FriendlyId
   friendly_id :name, use: :slugged
 
+  def self.by_popularity
+    select("categories.*, sum(order_items.quantity) as sales")
+    .left_joins(items: :order_items)
+    .group("categories.id", "items.id")
+    .distinct
+    .order("sales DESC")
```
Comment (jmejia) [inline app/models/category.rb:14]: 👍

File: app/models/item.rb [hunk 4]
```diff
@@ -10,4 +10,13 @@ class Item < ApplicationRecord
 
   belongs_to :store
   enum status: ["active", "retired"]
+
+  def self.by_popularity
+    select("items.*, count(order_items.id) as sales")
+    .left_joins(:order_items)
+    .group(:id)
+    .distinct
+    .order("sales DESC")
```
Comment (jmejia) [inline app/models/item.rb:19]: 👍

File: app/models/order.rb [hunk 5]
```diff
@@ -11,9 +11,8 @@ class Order < ApplicationRecord
   def add(item_hash)
     item_hash.each do |item, item_quantity|
       items << item
-      order_item = OrderItem.find_by(item_id: item.id)
-      order_items.find_by(item_id: item.id).update(quantity: item_quantity)
-      order_items.find_by(item_id: item.id).update(line_item_total: (item.price * item_quantity))
+      item.order_items.update(quantity: item_quantity)
+      item.order_items.update(line_item_total: (item.price * item_quantity))
```
Comment (jmejia) [inline app/models/order.rb:15]: Can we do this in one query? This is going to make multiple calls.

File: coverage/.last_run.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 87.77
+    "covered_percent": 87.83
```
Comment (jmejia) [inline coverage/.last_run.json:3]: Let's add the coverage directory to the gitignore. This is a lot to sift through and isn't really relevant to the code being committed.

**Review by jmejia — COMMENTED** [review]
Seems like there should be more tests for the amount of code submitted. I could be wrong on that, especially if some of this already had tests.

**Comment by vaidashi** [comment]
contained some of the old seed conflict data, will close this request

**Pull request closed without merging**